test/app/controllers/users.test.js

```diff
@@ -19,102 +38,241 @@ describe('POST /users', () => {
-        user.email += '@wolox.com.ar';
-        const res = await request(app)
-          .post('/users')
-          .send({
-            first_name: user.firstName,
-            last_name: user.lastName,
-            email: user.email,
-            password: user.password
-          });
-        expect(res.statusCode).toEqual(400);
-        expect(res.body.message).toEqual(failureMessage);
-        expect(res.body.internal_code).toEqual(internalCode);
+      it('should have status code 400', () => {
+        expect(shortPasswordResponse.statusCode).toEqual(400);
+      });
+      it('should respond with short password error message', () => {
+        expect(shortPasswordResponse.body.message).toEqual(
+          userSignUpErrorsMessages.wrongPasswordErrorMessage
+        );
+      });
+      it('should respond with validation error internal code', () => {
+        expect(shortPasswordResponse.body.internal_code).toEqual(validationErrorCode);
+      });
+    });
+
+    describe('With repeated email address', () => {
+      it('should have status code 422', () => {
+        expect(repeatedEmailResponse.statusCode).toEqual(422);
+      });
+      it('should respond with repeated email error message', () => {
+        expect(repeatedEmailResponse.body.message).toEqual(repeatedEmailErrorMessage);
+      });
+      it('should respond with user creation internal code', () => {
+        expect(repeatedEmailResponse.body.internal_code).toEqual(userCreationErrorCode);
+      });
+    });
+  });
+});
+
+describe('POST /users/sessions', () => {
+  let successResponse = {};
+  let emptyParamsResponse = {};
+  let validationErrorCode = {};
+  let wrongEmailResponse = {};
+  let wrongPasswordResponse = {};
+  let userLoginErrorCode = {};
+  beforeAll(async () => {
+    await factory.create('users', {
+      password: bcrypt.hash('passWord58', saltRounds),
+      email: 'fake@wolox.com.ar'
+    });
+    successResponse = await request(app)
+      .post('/users/sessions')
+      .send({
+        email: 'fake@wolox.com.ar',
+        password: 'passWord58'
+      });
+    validationErrorCode = 'validation_error';
+    emptyParamsResponse = await request(app)
+      .post('/users/sessions')
+      .send({
+        email: '',
+        password: ''
+      });
+    wrongEmailResponse = await request(app)
+      .post('/users/sessions')
+      .send({
+        email: 'fake@gmail.com.ar',
+        password: 'passWord58'
+      });
+    wrongPasswordResponse = await request(app)
+      .post('/users/sessions')
+      .send({
+        email: 'fake@wolox.com.ar',
+        password: 'wrongPassword'
+      });
+    userLoginErrorCode = 'user_login_error';
+  });
+  describe('Success case', () => {
+    describe('With valid parameters', () => {
+      it('should have status code 200', () => {
+        expect(successResponse.statusCode).toEqual(200);
+      });
+    });
+  });
+
+  describe('Failure cases', () => {
+    describe('With empty params', () => {
```

Comment:
Be more descriptive with the test. For example `FAIL because the body it's empty`.
Check this with all it descriptions